lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -200,13 +201,16 @@ const void *RequirementSource::getOpaqueStorage3() const {
   return nullptr;
 }
 
-bool RequirementSource::isInferredRequirement() const {
+bool RequirementSource::isInferredRequirement(bool includeQuietInferred) const {
   for (auto source = this; source; source = source->parent) {
     switch (source->kind) {
     case Inferred:
     case InferredProtocolRequirement:
       return true;
 
+    case QuietlyInferred:
+      return includeQuietInferred;
```

Comment:
Shouldn't the `includeQuietInferred == false` case keep walking up the chain, since it might be something like `Inferred -> QuietInferred`?